snapcraft/plugins/v2/python.py

```diff
@@ -106,14 +106,17 @@ def get_build_packages(self) -> Set[str]:
 
     def get_build_environment(self) -> Dict[str, str]:
         return {
+            # Add PATH to the python interpreter we always intend to use with
+            # this plugin. It can be user overridden, but that is an explicit
+            # choice made by a user.
+            "PATH": "${SNAPCRAFT_PART_INSTALL}/bin:${PATH}",
```

Comment:
I'd suggest changing SNAPCRAFT_PYTHON_INTERPRETER rather than modifying the path special for this plugin. Changing PATH will affect other things...

Reply:
The side effect of setting PATH is that we no longer need the venv, but the real reason is for override-build to work with what would pip install in $SNAPCRAFT_PART_INSTALL.